### ISSUE TITLE
Binding to polymorphic inbound request object

### ISSUE THREAD

**Issue by nhwilly** [FastEndpoints/FastEndpoints#856]
Just a question here.  I am trying to create a single endpoint for a polymorphic request.  It's not binding because it can't deserialize the request.  I can try to write a custom converter, but I don't understand why what I'm doing isn't working.  Especially since I thought I had it at one point and now tests are failing.  (I might have that wrong.)

I am getting this error: 
`{"Unable to instantiate type without a constructor! Offender: [Users.Client.Contracts.NotificationChannels.CreateChannelRequestBase]"}` 

in the `FastEndpoints.RequestBinder`1`

I thought STJ would manage this, but evidently not.  When I serialize/deserialize it in other spots, it's working.  That may be because in those cases I am informing the serializer/deserializer of the base type.

The question I have is: Do I have to create a custom converter for this ([which you have thoughtfully covered in the docs](https://fast-endpoints.com/docs/model-binding#custom-converters))?  Or have I missed something?

Here's the code - pretty simple, really:
```csharp
namespace Users.Client.Contracts.NotificationChannels;
[JsonDerivedType(typeof(CreateEmailChannel), typeDiscriminator: nameof(CreateEmailChannel))]
[JsonDerivedType(typeof(CreateSmsChannel), typeDiscriminator: nameof(CreateSmsChannel))]
[JsonDerivedType(typeof(CreatePushChannel), typeDiscriminator: nameof(CreatePushChannel))]

public abstract record CreateChannelRequestBase : RequestBase
{
  public required Guid Id { get; init; }
  public required string Name { get; init; }
  public const string RouteTemplate = "/users/me/NotificationChannels";
  public override string Route => RouteTemplate;
}


public record CreateEmailChannel : CreateChannelRequestBase
{
  public required string Email { get; init; }
}
public record CreateSmsChannel : CreateChannelRequestBase
{
  public required string CountryCode { get; init; }
  public required string PhoneNumber { get; init; }
}
public record CreatePushChannel : CreateChannelRequestBase
{
  public required string Email { get; init; }
  public required string ChannelId { get; init; }
  public required PushChannelProviderType PushChannelProviderType { get; init; }
}
```
And here's the endpoint:
```csharp
internal class CreateChannel(IMediator mediator) : Endpoint<CreateChannelRequestBase, Result<NotificationChannelDtoBase>>
{
  public override void Configure()
  {
    Post(CreateChannelRequestBase.RouteTemplate);
  }

  public override async Task<Result<NotificationChannelDtoBase>> ExecuteAsync(CreateChannelRequestBase request, CancellationToken ct = default)
  {
    var command = request.ToCommand();
    var result = await mediator.Send(command, ct);
    return result.Map(r => r.ToDto());
  }
}
```

**Comment by nhwilly** [comment]
Any chance this changed in FE recently?

**Comment by nhwilly** [comment]
Oooh.  Just found this. #172 

Maybe endpoint custom model binding is better?

**Comment by dj-nitehawk** [comment]
can you pls check the following repro project? seems to be working fine...

[Repro.zip](https://github.com/user-attachments/files/18282478/Repro.zip)

run the app and execute `request.http`

**Comment by dj-nitehawk** [comment]
added a test to the repro. works as expected too.

[Repro.zip](https://github.com/user-attachments/files/18282617/Repro.zip)

**Comment by nhwilly** [comment]
Over the top support.   🥇 

Thank you.  I was mostly just asking for advice. 

I'll check it out.  In the meantime, I used the custom binder on the endpoint and it worked great.

I'll be looking into Mr. Repro after I finish addressing these holiday cards...

**Comment by nhwilly** [comment]
OK.  I thought you were going to show me the binder, not that my code is broken.  😄 

I'll try to track it down.  I've done something to break it.  Thank you.

**Comment by dj-nitehawk** [comment]
i thought you put `abstract` there by mistake when posting here, because rider doesn't even let me build the project with `abstract` there.

**Comment by nhwilly** [comment]
Well that's weird.

Must be something we're both missing.  

The abstract _record_ - in this case - can't be instantiated without a type parameter.  It seems the JSON deserializer doesn't know what to do with it because it's not looking for the discriminator since it doesn't know about it.

I tried JSON constructor attributes, etc.  I couldn't get it to work.  I know at one point it did, but I've made a ton of sweeping changes ("I'll just clean this up right here...") that broke everything (rookie move) before I started testing again and who knows how much crap I screwed up.

In any case, the endpoint binder works well.  Glad you were foresighted enough to put it in.

Seriously, this is one of the best packages I've ever used.
